ui/apps/platform/src/hooks/useURLParameter.ts

```diff
@@ -32,7 +35,7 @@ function useURLParameter(keyPrefix: string, defaultValue: QueryValue): UseURLPar
     // memoize the setter function to retain referential equality as long
     // as the URL parameters do not change
     const setValue = useCallback(
-        (newValue: QueryValue) => {
+        (newValue: QueryValue, historyAction: Action = 'replace') => {
```

Comment:
It seems that we would want the default action to be `push`, so that the browser's forward and back buttons would allow you to "page" through your different searches.

I think "replace" may have been an inadvertent choice originally.

@stackrox/ui Can anyone think of a reason we might need `replace` as default?

Reply:
I think I went with replace as the default due to some of the first pages I converted behaving in that way (likely due to them using Redux instead of URL params). Working on "Risk" today I saw that the "push" action was pretty convenient and was thinking of proposing the same question. 😁

Reply:
My vote is to go with "push" behavior from here on out, unless there are extenuating circumstances.

Reply:
Yes, I agree with push.

Brainstorming about browser page title: Whatever, Whatever, Whatever, …
Is there any benefit to include in page title:
* search filter **categories** but **not values**
* sorting
* pagination

Reply:
Thinking about optional versus required function argument
* agree with **optional**: because consensus is for **push** usually or always instead of **replace** rarely or never
* instead of **required**: for visibility at call sites, if we thought that use cases were half and half

Reply:
Good point @pedrottimark. Also of note, changing the default behavior to 'push' will change how the current Dashboard, Violations, Clusters, Policies, and Risk Acceptance pages function. Ideally this is a positive change but I wanted to call it out since it is different from what is currently done.

(Possible that the reason this wasn't done before was because search was stored in Redux, which would have required a lot of boilerplate to implement "back" behavior with the search filters...)